Makefile

```diff
@@ -0,0 +1,61 @@
+MAKEFLAGS += --no-builtin-rules
+MAKEFLAGS += --no-builtin-variables
+
+# allow overriding the name of the venv directory
+VENV_DIR ?= venv
+
+# make tasks run all commands in a single shell
+.ONESHELL:
+
+# these targets don't produce files:
+.PHONY: clean filecheck pytest pytest-nb tests rerun-notebooks precommit-install precommit black pyright
+
+# remove all caches and the venv
+clean:
+	rm -rf ${VENV_DIR} .pytest_cache *.egg-info .coverage.*
+
+# run filecheck tests
+filecheck:
+	lit -vv tests/filecheck
+	lit -vv docs/Toy/examples
+
+# run pytest tests
+pytest:
+	pytest tests -W error -vv
+
+# run pytest on notebooks
+pytest-nb:
+	pytest -W error --nbval -vv docs --ignore=docs/mlir_interoperation.ipynb
+
+# run all tests
+tests: pytest pytest-nb filecheck pyright
+	echo test
+
+# re-generate the output from all jupyter notebooks in the docs directory
+rerun-notebooks:
+	jupyter nbconvert --ClearMetadataPreprocessor.enabled=True --inplace --to notebook --execute docs/*.ipynb
+
+# set up all precommit hooks
+precommit-install:
+	pre-commit install
+
+# run all precommit hooks and apply them
+precommit:
+	pre-commit run --all
+
+# run pyright on all files in the current git commit
+pyright:
+	pyright $(shell git diff --staged --name-only)
+
+# run black on all files currently staged
+black:
+	staged_files="$(shell git diff --staged --name-only)"
+	# run black on all of xdsl if no staged files exist
+	black $${staged_files:-xdsl}
```

Comment:
This also runs black on files that are not Python files. This is not an issue in the pyright case above, since pyright realises that and still works, but black complains.

Not sure we really care, just another minor inconvenience

Reply:
I have not cared so far, maybe we can fix that in an upcoming PR?